injectproxy/routes.go

```diff
@@ -118,27 +120,65 @@ func (r *routes) noop(w http.ResponseWriter, req *http.Request) {
 }
 
 func (r *routes) query(w http.ResponseWriter, req *http.Request) {
-	expr, err := parser.ParseExpr(req.FormValue("query"))
+	// The `query` can come in the URL query string and/or the POST body.
+	// For this reason, we need to try to enforcing in both places.
+	// Note: a POST request may include some values in the URL query string
```

Comment:
That sounds wrong, shouldn't we just expect one of ways?

Reply:
I think historically we once did but then someone used did it that other way and it broke. We definitely need to support both. :man_shrugging:

Reply:
This is literally how it works today in Prometheus because Prometheus uses r.FormValue

Reply:
Not really:

> Request body parameters take precedence over URL query string values in r.Form.

So there is no point in doing URL injection if there is form, we can just delete URL one if we want to be sure nothing leaks.

Reply:
Why? This is making assumptions about what the user wants. The user meant to send a query parameter? Great then escape that. The user also sent a form parameter? Great, then escape that. Otherwise you are mixing the semantics of the implementation detail of Go-s form parsing with business-logic just because both this proxy and Prometheus happen to be both written in Go. Escaping both ensures nothing leaks and that we don't assume one over the other.

Reply:
I think that a little bit more complex then it needs to be, but as long as safe - ok (: